### PR TITLE
feat(tui): custom slash commands from .stakpak/commands/ + /review command

### PR DESCRIPTION
**Pull request by mstfash**
## Summary

- Users can now create custom slash commands by placing `.md` files in `.stakpak/commands/` (project-local) or `~/.stakpak/commands/` (global). Each file becomes a `/filename` command available in the dropdown.
- Adds a built-in `/review` command that reviews code changes (uncommitted, a specific commit, branch, or PR).
- Fixes `build_review_command` which previously did not compile due to unescaped inner quotes and a missing return statement.

## How Custom Commands Work

**Create a file:**
```
.stakpak/commands/audit.md
```

**With optional YAML front matter for a description:**
```markdown
---
description: Run a security audit on the project
---

Perform a comprehensive security audit of this project.
Check for dependency vulnerabilities, hardcoded secrets, ...
```

If no front matter is provided, the first line of the file is used as the description (truncated to 60 chars).

**Use it:**
1. Type `/` — the dropdown shows all commands, custom ones prefixed with `[custom]`
2. Select `/audit` and press Enter or Tab — autocompletes to `/audit ` with trailing space
3. Optionally type extra context: `/audit focus on auth module`
4. Press Enter — sends the file content + extra text as a user message

**Priority:** Project-local (`.stakpak/commands/`) wins over global (`~/.stakpak/commands/`) on name clash. Built-in commands always take precedence — a custom `help.md` cannot override `/help`.

**Hot-reload:** Custom commands are re-scanned from disk each time the user types `/` to open the dropdown. No restart needed.

## `/review` Command

A new built-in slash command for code review:

- `/review` or `/review ` — reviews all uncommitted + staged changes
- `/review abc123` — reviews a specific commit
- `/review main` — diffs current branch against main
- `/review https://github.com/.../pull/42` — reviews a PR via `gh`

The review prompt instructs the agent to gather full file context (not just diffs), check for bugs, structure, performance, and behavioral changes, and use available tools (explore, search) to verify findings before flagging them.

### New file: `tui/src/services/custom_commands.rs`
Core module that scans `~/.stakpak/commands/` and `.stakpak/commands/` for `.md` files, parses optional YAML front matter for descriptions, validates filenames (alphanumeric + hyphens + underscores), and returns `Vec<HelperCommand>` with `CommandSource::Custom`. Includes 12 unit tests covering front matter parsing, edge cases (dashes in values, unclosed delimiters, empty files), description truncation, and filename validation.

### Modified: `tui/src/app/types.rs`
- `HelperCommand` fields changed from `&'static str` to `String` (required for runtime-loaded custom commands)
- Added `CommandSource` enum: `BuiltIn` for hardcoded commands, `Custom { prompt_content }` for user-defined ones

### Modified: `tui/src/services/commands.rs`
- All 23 built-in `HelperCommand` entries updated with `source: CommandSource::BuiltIn`
- `CommandId` type loosened from `&'static str` to `&'a str`
- Added `/review` to the helper command list and `execute_command()` match
- `/review` handler: autocompletes on exact `/review` input, fires immediately on `/review ` or `/review <args>`
- `execute_command()` fallback (`_ =>`) now checks `state.helpers` for custom commands, parses extra text, and sends `prompt_content + extra` as a user message
- Fixed `build_review_command`: uses `r#"..."#` raw string (inner quotes were breaking compilation), accepts `user_input: &str` parameter (was unused `state: &AppState`), properly returns `String` (was binding to `let` without returning)

### Modified: `tui/src/app.rs`
- `get_helper_commands()` merges built-in commands with custom commands loaded from disk, skipping name clashes

### Modified: `tui/src/services/handlers/input.rs`
- Hot-reloads `state.helpers` via `AppState::get_helper_commands()` when input is exactly `/` (single filesystem scan per dropdown open, not per keystroke)
- Enter from dropdown on custom commands autocompletes into input (doesn't fire immediately), letting user append extra context
- Defense-in-depth block extended to handle custom commands with arguments (e.g., `/audit focus on auth` pasted into input)

### Modified: `tui/src/services/handlers/misc.rs`
- Tab completion adds trailing space for `/review` and all custom commands (`CommandSource::Custom`)

### Modified: `tui/src/services/helper_dropdown.rs`
- Custom commands show `[custom]` prefix before their description in the dropdown

### Modified: `tui/src/services/helper_block.rs`
- Updated help message rendering for `String` fields (`.clone()` instead of implicit `Copy`)

### Modified: `tui/src/services/mod.rs`
- Added `pub mod custom_commands;`

## Testing

- `cargo check` — clean
- `cargo clippy --all-targets` — zero warnings
- `cargo fmt --check` — clean
- `cargo test -p stakpak-tui --lib` — 246 tests pass (12 new in `custom_commands`)

New tests cover:
- Front matter extraction (with/without description, quoted values, unclosed delimiters, empty body)
- `---` in description values doesn't break parsing (line-start matching)
- `---` in body content preserved correctly
- Filename validation (alphanumeric, hyphens, underscores; rejects spaces, dots, slashes)
- Empty file skipping
- Description truncation for long first lines

### CODE REVIEW

File: tui/src/services/handlers/input.rs [hunk 1]
```diff
@@ -485,6 +506,8 @@ fn handle_input_submitted(
             Some("/editor")
         } else if input.starts_with("/toggle_auto_approve ") {
             Some("/toggle_auto_approve")
+        } else if input.starts_with("/review ") {
+            Some("/review")
         } else {
             None
         };
```
Comment (ahmedhesham6) [inline tui/src/services/handlers/input.rs]: Can we match here?

File: tui/src/app/types.rs [hunk 2]
```diff
@@ -105,10 +105,21 @@ pub struct FileSearchResult {
...
 
+/// Source of a slash command — built-in or loaded from a custom `.md` file.
+#[derive(Debug, Clone, PartialEq)]
+pub enum CommandSource {
+    /// Hard-coded command handled by `execute_command()`.
+    BuiltIn,
+    /// User-defined command loaded from `~/.stakpak/commands/` or `.stakpak/commands/`.
+    /// The `prompt_content` is the raw markdown body of the file.
+    Custom { prompt_content: String },
+}
```
[the remark this answers is not in the document]
Comment (ahmedhesham6) [reply tui/src/app/types.rs]: So when matching we dont need to add a lot of steps just add the BulitInWithPrompt and the prompt file and it will work nativly
